app/models/calculator/default_tax.rb

```diff
@@ -13,6 +13,8 @@ def compute(computable)
       case computable
       when Spree::Order
         compute_order(computable)
+      when Spree::Shipment
+        compute_shipment(computable)
```

Comment:
I think it's time we replace this growing `case/when` with polymorphism. I believe this would make it much more changeable and thus more robust because the change would be scoped to the particular computable object and not this class. As far as I see, `#compute_order` doesn't depend on anything from this class and it passes `order` to the methods it cares about.

A rough potential solution would look similar to this :point_down: 

```rb
def compute(computable)
  computable.calculate_default_tax
end
```
:point_up: note `computable` can be any object that implements `#calculate_default_tax`. At this point we only need to trust that object. Then, we could move the logic to calculate the default tax in each case (Order, Shipment or LinteItem) to classes that only do that. So what if we do:

```rb
computable = OrderDefaulTax.new(order, calculable, EnterpriseFeeCalculator.new)

class OrderDefaultTax
  def initialize(order, rate, calculator)
    @order = order
    @rate = rate # This maps to the Spree::Calculator#calculable
    @calculator = calculator # EnterpriseFeeCalculator instance injected from the caller
  end

  def calculate_default_tax
    [
      line_items_total(order),
      per_item_fees_total(order, calculator),
      per_order_fees_total(order, calculator)
    ].sum do |total|
      round_to_two_places(total * rate.amount)
    end
  end

  private

  # All the helper methods used above go here
end
```

I believe this is the right time to explore this. Having a better design will help us reduce the cost of changing adjustments further. Fancy trying it out? I can help you.

Reply:
> I see shipments and line items share the same logic. We can do that by reusing the same object when calling #compute in those two cases

I already have another branch that does exactly that! :sweat_smile:  I haven't pushed it yet but it's already done :ok_hand: 

All of the super crazy code in this class (everything in `#compute_order`, and _all_ of the private methods used by `#compute_order`) is exactly what we are moving away from. It's only used with additional taxes (US and CA), and we definitely will _not_ be using it in the near future. :100:

Can we wait a little longer? This PR is already huuuuge, and the next two PRs after this one will be addressing this and moving us towards the point where we can just :fire: all these nasty bits.